src/ert/config/workflow_job.py

```diff
@@ -22,40 +24,74 @@
 ContentTypes: TypeAlias = type[int] | type[bool] | type[float] | type[str]
 
 
+class ErtScriptLoadFailure(ValueError):
+    pass
+
+
 def workflow_job_parser(file: str) -> ConfigDict:
     schema = init_workflow_job_schema()
     return parse(file, schema=schema)
 
 
-class ErtScriptLoadFailure(ValueError):
-    pass
+def workflow_job_from_file(config_file: str, name: str | None = None) -> WorkflowJob:
+    if not name:
+        name = os.path.basename(config_file)
+
+    content_dict = workflow_job_parser(config_file)
+    arg_types_list = WorkflowJob._make_arg_types_list(content_dict)
+    min_args = content_dict.get("MIN_ARG")  # type: ignore
+    max_args = content_dict.get("MAX_ARG")  # type: ignore
+    script = str(content_dict.get("SCRIPT")) if "SCRIPT" in content_dict else None  # type: ignore
+    internal = (
+        bool(content_dict.get("INTERNAL")) if "INTERNAL" in content_dict else None  # type: ignore
+    )
+    if internal is False:
+        ConfigWarning.deprecation_warn(
+            "INTERNAL FALSE has no effect and can be safely removed",
+            content_dict["INTERNAL"],  # type: ignore
+        )
+    if script and not internal:
+        ConfigWarning.deprecation_warn(
+            "SCRIPT has no effect and can be safely removed",
+            content_dict["SCRIPT"],  # type: ignore
+        )
+    if script is not None and internal:
+        msg = f"Deprecated keywords, SCRIPT and INTERNAL, for {name}, loading script {script}"
+        logger.warning(msg)
+        ConfigWarning.deprecation_warn(msg, content_dict["SCRIPT"])  # type: ignore
+        try:
+            ert_script = ErtScript.loadScriptFromFile(script)
+        # Bare Exception here as we have no control
+        # of exceptions in the loaded ErtScript
+        except Exception as err:
+            raise ErtScriptLoadFailure(f"Failed to load {name}: {err}") from err
+        return ErtScriptWorkflow(
+            ert_script=ert_script,
+            name=name,
+            min_args=min_args,
+            max_args=max_args,
+            arg_types=arg_types_list,
+            stop_on_fail=content_dict.get("STOP_ON_FAIL"),  # type: ignore
+        )
+    else:
+        return ExecutableWorkflow(
+            name=name,
+            min_args=min_args,
+            max_args=max_args,
+            arg_types=arg_types_list,
+            executable=content_dict.get("EXECUTABLE"),  # type: ignore
+            stop_on_fail=content_dict.get("STOP_ON_FAIL"),  # type: ignore
+        )
 
 
 @dataclass
 class WorkflowJob:
```

Comment:
Should we make this abstract base class private?
```suggestion
class _WorkflowJob:
```